### PR TITLE
Improved design documentation of tablet server

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: _docs-2-0/getting-started/design.md [hunk 1]
```diff
@@ -112,10 +112,14 @@ called a minor compaction. A new MemTable is then created and the fact of the
...
 
 When a request to read data arrives at a TabletServer, the TabletServer does a
-binary search across the MemTable as well as the in-memory indexes associated
-with each RFile to find the relevant values. If clients are performing a scan,
-several key-value pairs are returned to the client in order from the MemTable
-and the set of RFiles by performing a merge-sort as they are read.
+binary search across the MemTable as well as the index blocks associated with each RFile
+to find the relevant values. If clients are performing a scan, several key-value pairs
+are returned to the client in order from the MemTable and data blocks of RFiles by performing
+a merge-sort as they are read. If [caching] is enabled for the table, any index or data
```
Comment (keith-turner) [inline _docs-2-0/getting-started/design.md]: I think `sorted merge` would be more accurate than `merge-sort`.  The data is already sorted and `merge-sort` is the name of an algorithm for sorting unsorted data.

File: _docs-2-0/getting-started/design.md [hunk 2]
```diff
@@ -112,10 +112,14 @@ called a minor compaction. A new MemTable is then created and the fact of the
...
+block is stored in the block cache to speed up future scans.
+
+![tablet server diagram]({{ site.url }}/images/docs/tablet_server.png)
```
Comment (keith-turner) [inline _docs-2-0/getting-started/design.md]: The diagram could show there are two caches, an index cache and a data cache.

File: _docs-2-0/getting-started/design.md (hunk 1 above)
Comment (mikewalch) [reply _docs-2-0/getting-started/design.md]: Should be fixed now.

File: _docs-2-0/getting-started/design.md (hunk 2 above)
Comment (mikewalch) [reply _docs-2-0/getting-started/design.md]: Updated diagram

**Pull request merged**